src/MainWindow.vala

```diff
@@ -1462,11 +1466,11 @@ namespace Terminal {
 
             foreach (TerminalWidget terminal in terms) {
                 string term_path = terminal.current_working_directory;
-                string term_label = Path.get_basename (term_path);
+                string term_label = terminal.window_title != "" ? terminal.window_title
+                                                                : Path.get_basename (term_path);
```

Comment:
This causes a regression when there *isn't* a custom prompt set in `.bashrc`.  In that case the terminal label is the same as the terminal prompt (and window title) which is not what is wanted.  The label should always show the terminal path (disambiguated).

Reply:
that's because `.bashrc` set a title by default:
```bash
# If this is an xterm set the title to user@host:dir
case "$TERM" in
xterm*|rxvt*)
    PS1="\[\e]0;${debian_chroot:+($debian_chroot)}\u@\h: \w\a\]$PS1"
    ;;
*)
    ;;
esac
```
elementary use a patch to set the title as `dir: last_cmd` or `actual_cmd`, see: https://github.com/elementary/os-patches/blob/712e7a310b32c205fde0c73025c54f1be9181e3e/debian/skel.bashrc#L114-L171